src/asl/synth.ts

```diff
@@ -1894,12 +1894,77 @@ export class ASL {
               return { value: !!argOutput.value, containsJsonPath: false };
             }
           });
+        } else if (ref === String) {
+          const [arg] = expr.args;
```

Comment:
Do you want to check for `String(undefined)`, i.e. where `expr.args` is an `UndefinedLiteralExpr`?

Reply:
Currently that will show `Step Functions does not support undefined`, but I can make it work at least.